### PR TITLE
Prefer system FindSDL_audiolib

### PR DESCRIPTION
**Pull request by SuperSandro2000**

### CODE REVIEW

File: CMake/finders/FindSDL_audiolib.cmake [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+find_library(SDL_audiolib_LIBRARY
+             NAMES SDL_audiolib libSDL_audiolib
+             HINTS ${PC_SDL_audiolib_LIBDIR} ${PC_SDL_audiolib_LIBRARY_DIRS})
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(SDL_audiolib DEFAULT_MSG
+                                  SDL_audiolib_INCLUDE_DIR SDL_audiolib_LIBRARY)
+
+if(SDL_audiolib_FOUND AND NOT TARGET SDL2::SDL_audiolib)
+  add_library(SDL2::SDL_audiolib UNKNOWN IMPORTED)
```
Comment (glebm) [inline CMake/finders/FindSDL_audiolib.cmake:16]: 1. `SDL_audiolib` is not a part of SDL2; The name should be `SDL_audiolib::SDL_audiolib`
2. In 3rdParty/SDL_audiolib/CMakeLists.txt, add `add_library(SDL_audiolib::SDL_audiolib ALIAS SDL_audiolib)`
3. Use `SDL_audiolib::SDL_audiolib` as the dependency name throughout.

File: CMake/Dependencies.cmake [hunk 2]
```diff
@@ -118,9 +118,19 @@ else()
...
+      message("-- Found SDL_audiolib")
+    else()
+      message("-- Suitable system SDL_audiolib package not found, will use SDL_audiolib from source")
+      set(DEVILUTIONX_SYSTEM_SDL_AUDIOLIB OFF)
+    endif()
+  endif()
+  dependency_options("SDL_audiolib" DEVILUTIONX_SYSTEM_SDL_AUDIOLIB ON DEVILUTIONX_STATIC_SDL_AUDIOLIB)
   if(DEVILUTIONX_SYSTEM_SDL_AUDIOLIB)
     find_package(SDL_audiolib REQUIRED)
+    include_directories(${SDL_audiolib_INCLUDE_DIR})
```
Comment (glebm) [inline CMake/Dependencies.cmake:133]: The imported package already defines `INTERFACE_INCLUDE_DIRECTORIES`, this is not necessary.

**Review by glebm — CHANGES_REQUESTED** [review]

**Comment by tsunamistate** [comment]
I don't know if this is needed.
Is SDL_Audiolib provided in any of Linux distributions as a package?

**Comment by SuperSandro2000** [comment]
> I don't know if this is needed. Is SDL_Audiolib provided in any of Linux distributions as a package?

Yes, it is by nixpkgs/NixOS

**Comment by SuperSandro2000** [comment]
Force pushed the requested changes. I hope I didn't miss anything. My CMake knowledge could be better.

File: CMake/Dependencies.cmake [hunk 3]
```diff
@@ -118,7 +118,16 @@ else()
 endif()
 
 if(NOT NOSOUND)
-  dependency_options("SDL_audiolib" DEVILUTIONX_SYSTEM_SDL_AUDIOLIB OFF DEVILUTIONX_STATIC_SDL_AUDIOLIB)
+  if(NOT DEFINED DEVILUTIONX_SYSTEM_SDL_AUDIOLIB)
+	find_package(SDL_audiolib QUIET)
```
Comment (glebm) [inline CMake/Dependencies.cmake:122]: This is a tab but should be 4 spaces

File: CMake/finders/FindSDL_audiolib.cmake [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+                        IMPORTED_LOCATION ${SDL_audiolib_LIBRARY}
+                        INTERFACE_INCLUDE_DIRECTORIES ${SDL_audiolib_INCLUDE_DIR})
+endif()
+
+if(SDL_audiolib_FOUND)
+  mark_as_advanced(SDL_audiolib_INCLUDE_DIR SDL_audiolib_LIBRARY)
+  set(SDL_audiolib_LIBRARIES ${SDL_audiolib_LIBRARY})
+  set(SDL_audiolib_INCLUDE_DIRS ${SDL_audiolib_INCLUDE_DIR})
+endif()
+
```
Comment (glebm) [inline CMake/finders/FindSDL_audiolib.cmake:27]: Extra blank line at the end of file

**Review by glebm — CHANGES_REQUESTED** [review]
Looks good overall apart from a couple of formatting issues

File: CMake/finders/FindSDL_audiolib.cmake (hunk 4 above)
Comment (AJenbo) [reply CMake/finders/FindSDL_audiolib.cmake:27]:
```suggestion
```

File: CMake/Dependencies.cmake (hunk 3 above)
Comment (AJenbo) [reply CMake/Dependencies.cmake:122]:
```suggestion
    find_package(SDL_audiolib QUIET)
```

**Pull request merged**